drivers/platform/stm32/stm32_uart.c

```diff
@@ -115,7 +131,44 @@ int32_t no_os_uart_init(struct no_os_uart_desc **desc,
 
 	*desc = descriptor;
 
+	// nonblocking uart_read
+	if(param->asynchronous_rx) {
+		ret = lf256fifo_init(&descriptor->rx_fifo);
+		if (ret < 0)
+			goto error;
+
+		struct no_os_irq_init_param nvic_ip = {
+			.platform_ops = &stm32_irq_ops,
+			.extra = sud->huart,
+		};
+		ret = no_os_irq_ctrl_init(&nvic, &nvic_ip);
```

Comment:
This would be a memory leak, but I think we shouldn't do initialization here at all because irq should be either initialized by the application or have the API know if someone already initialized it and return the handler without changing what was done so far.

Reply:
I rewrote a bit this part, moving the irq controller descriptor to the stm32_uart_desc.
If you check the code now, there shouldn't be any risk of memory leak.